src/components/Report.tsx

```diff
@@ -108,6 +152,84 @@ export const Report = ({
     setFocusedWorkflows([...allWorkflows, 'Unorganized']);
   }, [workflowMapping]);
 
+  useEffect(() => {
+    data?.filerecords?.map((filterRecord: any) => {
+      issueCount = issueCount + filterRecord.auditrecords.length;
+      filterRecord.auditrecords.map((auditRecord: any) => {
+        const issueId = auditRecord.auditinfo.issueid;
+        if (hasHelpArticle(issueId)) issueLinkedCount += 1;
+      });
+    });
+
+    valueRef.current = {
+      issueCount: issueCount,
+      linkedIssueCount: issueLinkedCount,
+    };
+  }, [data]);
+
+  const triggerEvent = (event: any) => {
+    triggerSyncReportOpenedEvent();
+    event.returnValue = '';
+  };
+
+  const triggerSyncReportOpenedEvent = useCallback(() => {
```

Comment:
I think it would be much cleaner to move all the things related to events reporting to another file. This component gets too big and does too many thing at one time.

Reply:
I will revisit this later, when we will push final changes.